projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,154 @@
+#include "ads1015.h"
+// All the important data about ADS1015 is stored in ADS1015Storage storage by different functions.
+// Ads1015_init writes the desired settings to registers and inits the GPIO pin (ALRT/RDY).
+// From there, the user would start configuring channels with ads1015_configure_channel, and inside
+// this function an interrupt is registered on the ALRT/RDY pin with prv_interrupt_handler. Once the
+// interrupt is raised, prv_interrupt_handler reads the conversion values from the registers and
+// stores them in storage. The user could access this data by the provided read functions. These
+// functions get their data from storage and not directly by reading the registers. Also look at
+// ads1015_def for more information on macros and how registers are setup.
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(Ads1015Storage *storage) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  return i2c_write(storage->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
```

Comment:
If you send the reset packet to the general call address, it resets all I2C slaves since all chips use the same general call address as far as I can tell. I'm not sure if that's something we really need to support.